clients/index.js

```diff
@@ -564,98 +514,130 @@ ApplicationClients.prototype.updateLazyHandling = function updateLazyHandling()
     }
 };
 
-ApplicationClients.prototype.updateReservoir = function updateReservoir() {
+ApplicationClients.prototype.updateReservoir = function updateReservoir(hasChanged, forceUpdate) {
     var self = this;
-    if (self.logReservoir) {
+
+    if (!self.logReservoir) {
+        return;
+    }
+
+    if (forceUpdate || hasChanged['log.reservoir.size']) {
         var size = self.remoteConfig.get('log.reservoir.size', 100);
-        var interval = self.remoteConfig.get('log.reservoir.flushInterval', 50);
+        self.logReservoir.setSize(size);
+    }
 
+    if (forceUpdate || hasChanged['log.reservoir.flushInterval']) {
+        var interval = self.remoteConfig.get('log.reservoir.flushInterval', 50);
         self.logReservoir.setFlushInterval(interval);
```

Comment:
@rf please let us know if setFlushInterval depends on a following setSize to effect the change.